stl/inc/algorithm

```diff
@@ -1786,6 +1786,39 @@ template <class _ExPo, class _FwdIt, class _Pr, _Enable_if_execution_policy_t<_E
 _NODISCARD _FwdIt remove_if(_ExPo&& _Exec, _FwdIt _First, const _FwdIt _Last, _Pr _Pred) noexcept; // terminates
 #endif // _HAS_CXX17
 
+// FUNCTION TEMPLATE _Erase_nodes_if

```

Comment:
I agree. Compiler throughput (i.e. build speed) is a major concern, and `<algorithm>` is one of our largest headers.

Reply:
Ah, I see the issue - `remove()` and `remove_if()` currently live in `<algorithm>` immediately above this diff, so that's why @SuperWig carefully put `_Erase_remove()` and `_Erase_remove_if()` here. I previously solved this by dragging `<algorithm>` into `<experimental/xutility>`, being oblivious to throughput (it is a common problem for me). Now that the container headers themselves are required to provide erasure, we do need a proper solution. Moving `remove()` and `remove_if()` into `<xmemory>` appears to be the right solution (note that `<algorithm>` includes `<xmemory>` so users of `<algorithm>` will be unaffected). We move certain algorithms even higher up to `<xutility>` (like `equal()`) but that isn't necessary here.

Reply:
Is there a specific place they should live in there? Nothing is striking me as an obvious place to put them.

Reply:
> Is there a specific place they should live in there? Nothing is striking me as an obvious place to put them.

We have tentative plans to split up and/or reorganize `<xutility>` and `<xmemory>` (and rename them with `__msvc` prefixes) sometime after the move from our internal repo to Github is complete, so I wouldn't worry about it too much.